### PR TITLE
lb-ipam,metrics: Added LB-IPAM metrics

### PR DESCRIPTION
**Pull request by dylandreimerink**
This PR adds metrics for the most critical parameters for users of the LB-IPAM feature.

```release-note
Add metrics for LB-IPAM
```

### CODE REVIEW

**Review by aspsk — APPROVED** [review]

File: Documentation/observability/metrics.rst [hunk 1]
```diff
@@ -548,6 +548,20 @@ Name                                     Labels
...
+LB-IPAM
+~~~~~~~
+
+======================================== ================================================================= ========== ========================================================
+Name                                     Labels                                                            Default    Description
+======================================== ================================================================= ========== ========================================================
+``lbipam_conflicting_pools_total``                                                                         Enabled    Number of conflicting pools
+``lbipam_ips_available_total``           ``pool``                                                          Enabled    Number of available IPs per pool.
+``lbipam_ips_used_total``                ``pool``                                                          Enabled    Number of used IPs per pool.
+``lbipam_services_matching_total``                                                                         Enabled    Number of matching services.
```
Comment (zacharysarah) [inline Documentation/observability/metrics.rst]: Keep periods consistent across similar items
```suggestion
``lbipam_ips_available_total``           ``pool``                                                          Enabled    Number of available IPs per pool
``lbipam_ips_used_total``                ``pool``                                                          Enabled    Number of used IPs per pool
``lbipam_services_matching_total``                                                                         Enabled    Number of matching services
```

**Review by zacharysarah — APPROVED** [review]
MInor nits, approving to unblock with the understanding that changes are required prior to merge.

**Review by aditighag — APPROVED** [review]
It would be useful to cross-reference the metrics in a lb-ipam specific (troubleshooting) doc.

**Pull request merged**